### PR TITLE
Remove unneeded null escaping

### PR DESCRIPTION
**Pull request by thegrb93**

### CODE REVIEW

**Comment by AbigailBuccaneer** [comment]
LGTM, feel free to merge after testing

**Comment by thegrb93** [comment]
I think there's one small bug. It's probably better to send the size as an int rather than doing the calculation anyway.

File: lua/advdupe2/cl_networking.lua [hunk 1]
```diff
@@ -271,8 +274,9 @@ local function SendFileToServer(eof)
 	AdvDupe2.ProgressBar.Percent = math.min(math.floor((AdvDupe2.LastPos/AdvDupe2.Length)*100),100)
 
 	net.Start("AdvDupe2_ReceiveFile")
-		net.WriteBit(AdvDupe2.LastPos>=AdvDupe2.Length)
-		net.WriteString(data)
+		net.WriteInt(AdvDupe2.LastPos>=AdvDupe2.Length and 1 or 0, 8)
+		net.WriteUInt(data, 32)
```
Comment (thegrb93) [inline lua/advdupe2/cl_networking.lua]: Should be #data

**Comment by thegrb93** [comment]
Need to make sure old files with null escaping still work.

**Comment by thegrb93** [comment]
File functions didn't need to be changed, but oh well. Also files are stored with nulls so no issue there.

**Comment by thegrb93** [comment]
Tested and working

**Pull request merged**